core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClient.java

```diff
@@ -270,11 +274,41 @@ protected static ClientRequestContext newDerivedContext(ClientRequestContext ctx
                                                             boolean initialAttempt) {
         final RequestId id = ctx.options().requestIdGenerator().get();
         final EndpointGroup endpointGroup = ctx.endpointGroup();
+        final ClientRequestContext derived;
         if (endpointGroup != null && !initialAttempt) {
-            return ctx.newDerivedContext(id, req, rpcReq, endpointGroup.select(ctx));
+            derived = ctx.newDerivedContext(id, req, rpcReq, endpointGroup.select(ctx));
         } else {
-            return ctx.newDerivedContext(id, req, rpcReq);
+            derived = ctx.newDerivedContext(id, req, rpcReq);
         }
+
+        final RequestLogAccess parentLog = ctx.log();
+        final RequestLog partial = parentLog.partial();
+        final RequestLogBuilder logBuilder = derived.logBuilder();
+        // serializationFormat is always not null, so this is fine.
+        logBuilder.serializationFormat(partial.serializationFormat());
+        if (parentLog.isAvailable(RequestLogProperty.NAME)) {
+            final String name = partial.name();
+            if (name != null) {
+                logBuilder.name(name);
+            }
+        }
+
+        final RequestLogBuilder parentLogBuilder = ctx.logBuilder();
+        if (parentLogBuilder.isDeferRequestContentSet()) {
+            logBuilder.deferRequestContent();
+        }
+        parentLog.whenAvailable(RequestLogProperty.REQUEST_CONTENT).thenApply(requestLog -> {
```

Comment:
I wonder if I have to copy the `RESPONSE_CONTENT` as well. 🤔

Reply:
I think so. How about copying `SESSION` and `SCHEME` as well?

Reply:
I think `SESSION` and `SCHEME` are not available at this point.
They are rather copied from child to parent.